canopen/pdo/base.py

```diff
@@ -4,59 +4,51 @@
 import logging
 import binascii
 
-from .sdo import SdoAbortedError
-from . import objectdictionary
-from . import variable
-
+from ..sdo import SdoAbortedError
+from .. import objectdictionary
+from .. import variable
 
 PDO_NOT_VALID = 1 << 31
 RTR_NOT_ALLOWED = 1 << 30
 
-
 logger = logging.getLogger(__name__)
 
 
-class PdoNode(collections.Mapping):
-    """Represents a slave unit."""
+class PdoBase(collections.Mapping):
+    """Represents the base implemention for the PDO object.
+    :param object node:
+        Parent object associated with this PDO instance
+    """
 
     def __init__(self, node):
+        """
+
+        """
         self.network = None
+        self.map = None
         self.node = node
-        self.rx = Maps(0x1400, 0x1600, self, 0x200)
-        self.tx = Maps(0x1800, 0x1A00, self, 0x180)
 
     def __iter__(self):
-        for pdo_maps in (self.rx, self.tx):
-            for pdo_map in pdo_maps.values():
-                for var in pdo_map.map:
-                    yield var.name
+        for var in self.map:
+            yield var.name
 
     def __getitem__(self, key):
-        for pdo_maps in (self.rx, self.tx):
-            for pdo_map in pdo_maps.values():
-                for var in pdo_map.map:
-                    if var.length and var.name == key:
-                        return var
-        raise KeyError("%s was not found in any map" % key)
+        if isinstance(key, int):
+            return self.map[key]
+        raise KeyError("PDO: {0} was not found in any map".format(key))
```

Comment:
The idea of being able to use a variable name as key is to allow the code to not care about in which PDO the variable happens to be mapped to.

Reply:
fix